subsys/net/ip/tcp.h

```diff
@@ -137,8 +140,10 @@ struct net_tcp {
 	 * of various timing issues when timer is scheduled to run.
 	 */
 	u32_t ack_timer_cancelled : 1;
+	/* Tells if fin timer has been already cancelled. */
+	u32_t fin_timer_cancelled : 1;
 	/** Remaining bits in this u32_t */
-	u32_t _padding : 12;
+	u32_t _padding : 11;
```

Comment:
Im not sure I understand what this is doing, but since this is not marked as packed struct doesn't the compiler would do the padding anyway? Btw it would probably be possible to use ATOMIC_DEFINE and migrate every field that require just 1 bit to be a flag, though this probably unrelated to this so it can be done later.